homeassistant/components/light/homekit_controller.py

```diff
@@ -0,0 +1,181 @@
+"""
+Support for Homekit lights.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.homekit/
+"""
+import colorsys
+import json
+import logging
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ATTR_RGB_COLOR, ATTR_COLOR_TEMP, SUPPORT_BRIGHTNESS,
+    SUPPORT_RGB_COLOR, SUPPORT_COLOR_TEMP, Light)
+
+REQUIREMENTS = ['homekit==0.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, service):
+    """Set up Homekit lighting."""
+    add_devices([HomekitLight(service)])
+
+
+class HomekitLight(Light):
+    """Representation of a Homekit light."""
+    def __init__(self, service):
+        """Initialize the light."""
+
+        self._name = service['device'].model
+        self._securecon = service['device'].securecon
+        self._aid = service['aid']
+        self._iid = service['service']['iid']
+        self._address = "%s-%s" % (service['device'].hkid, service['aid'])
```

Comment:
This is used as the unique_id
Is this right? Looking at the Homekit Accessory Protocol I would expect the unique ID to be comprised of the Accessory ID (`aid`?) and the Service or Characteristic ID (`iid`?)
As a single accessory (`aid`) can expose multiple services (light, fan, sensor) at least the `service['iid']` should be included right in the unique_id right?

It appears that `aid` and `iid` can change with a new pairing.  I don't know what `service['device'].hkid` returns. But if that is truly unique and always the same (Mac-address maybe) then it should be enough to just base the unique_id on the `hkid` and the `service['iid']`

Please correct me if I'm wrong :)

Reply:
Hm, yes - using aid isn't sufficient here, a single accessory can certainly expose multiple lights. Each service should also have a unique serial number, so with luck that'll be stable over pairings.